File: SPECIAL DAY/PYTHON/emeka_quiz_grading_system.py
num_students = int(input("Enter number of students: "))
num_quizzes = int(input("Enter number of quizzes: "))



scores = []


for index in range(num_students):
    print("--- Student", (index + 1), "---")
    student_scores = []
    
    for counter in range(num_quizzes):
        
        score = float(input("Score for Quiz " + str(counter + 1) + ": "))
        while score < 0 or score > 100:
            print("Invalid score! Must be between 0 and 100.")
            score = float(input("Score for Quiz " + str(counter + 1) + ": "))
            
        student_scores.append(score)
    scores.append(student_scores)
    print()

print("============= QUIZ GRADE REPORT =============")
print("STUDENT\t\tSCORES\t\tAVG")

for index in range(num_students):
    total_score = 0
    
    for score in scores[index]:
        total_score += score
        
    student_avg = total_score / num_quizzes
    print("Student", (index + 1), "\t", scores[index], "\t", student_avg)

print("---------------------------------------------")


best_quiz = 0
highest_avg = 0.0

print("Quiz averages: ", end="")

for counter in range(num_quizzes):
    quiz_total = 0
    
    for index in range(num_students):
        quiz_total += scores[index][counter]
        
    quiz_avg = quiz_total / num_students
    print("QZ" + str(counter + 1) + ":", quiz_avg, "  ", end="")
    
    
    if quiz_avg > highest_avg:
        highest_avg = quiz_avg
        best_quiz = counter + 1

print()
print("Best quiz: Quiz", best_quiz, "(avg", highest_avg, ")")

File: SPECIAL DAY/JAVA/EmekaQuizGradingSystem.java
import java.util.Scanner;

public class EmekaQuizGradingSystem {

public static void main(String[] args) {
Scanner scanner = new Scanner(System.in);

        
System.out.print("Enter number of students: ");
int numStudents = scanner.nextInt();
        
System.out.print("Enter number of quizzes: ");
int numQuizzes = scanner.nextInt();
System.out.println();

        
double[][] scores = new double[numStudents][numQuizzes];

        
for (int count = 0; count < numStudents; count++) {
    System.out.println("--- Student " + (count + 1) + " ---");
for (int counter = 0; counter < numQuizzes; counter++) {
                
    System.out.print("Score for Quiz " + (counter + 1) + ": ");
    double score = scanner.nextDouble();
                
                
while (score < 0 || score > 100) {
    System.out.println("Invalid score! Must be between 0 and 100.");
    System.out.print("Score for Quiz " + (counter + 1) + ": ");
    score = scanner.nextDouble();
    }
    scores[count][counter] = score;
    }
    System.out.println();
    }


System.out.println("============= QUIZ GRADE REPORT =============");
        
        
for (int count = 0; count < numStudents; count++) {
    System.out.print("Student " + (count + 1) + ": \t");
    double studentSum = 0;
            

for (int counter = 0; counter < numQuizzes; counter++) {
    System.out.print(scores[count][counter] + "\t");
    studentSum += scores[count][counter];
    }
            
    double studentAvg = studentSum / numQuizzes;
    System.out.println(" | Avg: " + studentAvg);
    }

System.out.println("---------------------------------------------");

      
int bestQuizNum = 1;
double highestQuizAvg = 0.0;

System.out.print("Quiz averages: ");
        
for (int counter = 0; counter < numQuizzes; counter++) {
    double quizSum = 0;
            

for (int count = 0; count < numStudents; count++) {
    quizSum += scores[count][counter];
    }
            
    double quizAvg = quizSum / numStudents;
    System.out.print("QZ" + (counter + 1) + ": " + quizAvg + "   ");
            
    if (quizAvg > highestQuizAvg) {
    highestQuizAvg = quizAvg;
    bestQuizNum = counter + 1;
    }
    }
        
    System.out.println();
    System.out.println("Best quiz: Quiz " + bestQuizNum + " (avg " + highestQuizAvg + ")");
        
      
    }
}
